src/system/PermissionsModule/Api/PermissionApi.php

```diff
@@ -0,0 +1,323 @@
+ *
+ * This work is contributed to the Zikula Foundation under one or more
+ * Contributor Agreements and licensed to You under the following license:
+ *
+ * @license GNU/LGPLv3 (or at your option, any later version).
+ *
+ * Please see the NOTICE file distributed with this source code for further
+ * information regarding copyright and licensing.
+ */
+
+namespace Zikula\PermissionsModule\Api;
+
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+use Zikula\PermissionsModule\Entity\RepositoryInterface\PermissionRepositoryInterface;
+use Zikula\UsersModule\Entity\RepositoryInterface\UserRepositoryInterface;
+use Zikula\UsersModule\Entity\UserEntity;
+
+/**
+ * PermissionApi.
+ */
+class PermissionApi
+{
+    /**
+     * 'all users', includes unregistered users
+     */
+    const ALL_USERS = -1;
+    /**
+     * unregistered user
+     */
+    const UNREGISTERED_USER = 0;
+    /**
+     * 'all groups', includes unregistered users
+     */
+    const ALL_GROUPS = -1;
+    /**
+     * unregistered users.
+     */
+    const UNREGISTERED_USER_GROUP = 0;
+
+    /**
+     * @var PermissionRepositoryInterface
+     */
+    private $permRepository;
+
+    /**
+     * @var UserRepositoryInterface
+     */
+    private $userRepository;
+
+    /**
+     * @var SessionInterface
+     */
+    private $session;
+
+    /**
+     * @var \Zikula\Common\Translator\Translator
+     */
+    private $translator;
+
+    /**
+     * holds the cache of group Perms by User
+     * @var array
+     */
+    private $groupPermsByUser = [];
+
+    /**
+     * PermissionApi constructor.
+     * @param PermissionRepositoryInterface $permRepository
+     * @param UserRepositoryInterface $userRepository
+     * @param SessionInterface $session
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(PermissionRepositoryInterface $permRepository, UserRepositoryInterface $userRepository, SessionInterface $session, TranslatorInterface $translator)
+    {
+        $this->permRepository = $permRepository;
+        $this->userRepository = $userRepository;
+        $this->session = $session;
+        $this->translator = $translator;
+    }
+
+    /**
+     * Check permissions
+     *
+     * @param string   $component Component.
+     * @param string   $instance  Instance.
+     * @param integer  $level     Level.
+     * @param integer  $user      User Id.
+     *
+     * @return boolean
+     */
+    public function hasPermission($component = null, $instance = null, $level = ACCESS_NONE, $user = null)
+    {
+        if (!is_numeric($level)) {
+            throw new \InvalidArgumentException('Invalid security level');
+        }
+        if (isset($user) and !is_numeric($user)) {
+            throw new \InvalidArgumentException('User argument must be an integer.');
+        }
+        if (!isset($user)) {
+            $user = $this->session->get('uid', false);
```

Comment:
Hmm, I will look as well. Questions to answer for Symfony:
- Is there a request object if the CLI is used?
- Is there a session object if the CLI is used?
- [How does this behave if called via CLI](http://symfony.com/doc/current/book/security.html#securing-controllers-and-other-code)?